Automattic/jetpack
# Options: Add wpcom_newsletter_categories_enabled site option #32569

Closes https://github.com/Automattic/wp-calypso/issues/80728

## Proposed changes:

* Adds `wpcom_newsletter_categories_enabled` site option.

### Other information:


## Jetpack product discussion
pdDOJh-2gO-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

* Apply the PR to your sandbox (see how to: PCYsg-eg0-p2)
* Sandbox `public-api.wordpress.com`
* Make a GET request to `https://public-api.wordpress.com/rest/v1.4/sites/{site-id}/settings`
* The `wpcom_newsletter_categories_enabled` should be present on the response body under `settings`
